pandas/core/apply.py

```diff
@@ -312,26 +318,36 @@ def agg_list_like(self) -> DataFrame | Series:
         results = []
         keys = []
 
-        # degenerate case
-        if selected_obj.ndim == 1:
-            for a in arg:
-                colg = obj._gotitem(selected_obj.name, ndim=1, subset=selected_obj)
-                new_res = colg.aggregate(a)
-                results.append(new_res)
+        is_groupby = isinstance(obj, (DataFrameGroupBy, SeriesGroupBy))
+        context_manager: ContextManager
+        if is_groupby:
```

Comment:
Kinda unfortunate that groupby knowledge is leaking in `pandas/core/apply.py`, guessing there's no way to avoid this?